books/RayTracingTheRestOfYourLife.html

```diff
@@ -3171,9 +3171,11 @@
             ray scattered = ray(rec.p, p.generate(), r.time());
             auto pdf_val = p.value(scattered.direction());
 
-            return emitted
-                + srec.attenuation * rec.mat->scattering_pdf(r, rec, scattered)
-                                   * ray_color(scattered, depth-1) / pdf_val;
+            color scattered_color = (srec.attenuation
+                * rec.mat->scattering_pdf(r, rec, scattered)
+                * ray_color(scattered, depth-1)) / pdf_val;
```

Comment:
What would you think about something like the following?

```
auto sample_color = ray_color(scattered, depth-1);
auto scattering_pdf = rec.mat->scattering_pdf(r, rec, scattered);
color scattered_color = (srec.attenuation * scattering_pdf * sample_color) / pdf_val;
```

Reply:
Let me think about this one. I've created a separate issue #989  to think about it.